design/036-package_id.md

```diff
@@ -0,0 +1,106 @@
+# Proposal: Conan 2.0 requirements effect in PackageID
+
+| **Status**        |                                                   |
+|:------------------|:--------------------------------------------------|
+| **RFC #**         | [036](https://github.com/conan-io/tribe/pull/36)  |
+| **Submitted**     | 2022-06-17                                        |
+| **Tribe votes**   |                                                   |
+
+
+## Summary
+This proposal is about the effect that the dependencies have over the current package ``package_id``, but not about other factors, such as the package settings, options, configuration or other type of binary compatibility resulting from other factors other than dependencies.
+
+Conan 2.0 proposes a new default ``package_id`` that will be dynamic, based on two main cases:
+
+- When an application is linking a static library, or a shared library is linking a static library, or any library or application is linking a header-only library as an implementation detail, the resulting artifacts will be fully embedding/inlining the dependency artifacts, and they require in the general case to be re-built from sources when something in the dependency changes. This will be the “embed” case, and by default will use a “full_mode” policy, which means that the full dependency reference ``pkg/version@user/channnel#recipe_revision:package_id`` will be used and factored in the consumer ``package_id``. Recall that ``package_revision`` can no longer be part of the ``package_id`` as removed in proposal https://github.com/conan-io/tribe/pull/30
```

Comment:
```suggestion
- When an application is linking a static library, or a shared library is linking a static library, or any compiled library or application is linking a header-only library as an implementation detail, the resulting artifacts will be fully embedding/inlining the dependency artifacts, and they require in the general case to be re-built from sources when something in the dependency changes. This will be the “embed” case, and by default will use a “full_mode” policy, which means that the full dependency reference ``pkg/version@user/channnel#recipe_revision:package_id`` will be used and factored in the consumer ``package_id``. Recall that ``package_revision`` can no longer be part of the ``package_id`` as removed in proposal https://github.com/conan-io/tribe/pull/30
```

One header-only library should be allowed to depend on another header-only library without triggering this.

The next paragraph acknowledges that with "or a header-only library is depending on another library" but as written the two rules conflict.

Reply:
You are right, better clarify that this is for compiled libraries.